test/core/security/local_security_connector_test.cc

```diff
@@ -0,0 +1,139 @@
+//
+//
+// Copyright 2024 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#include <gtest/gtest.h>
+
+#include <grpc/impl/grpc_types.h>
+
+#include "src/core/client_channel/client_channel_filter.h"
+#include "src/core/lib/security/context/security_context.h"
+#include "src/core/tsi/transport_security.h"
+#include "test/core/test_util/test_config.h"
+
+namespace grpc_core {
+namespace testing {
+namespace {
+
+absl::string_view GetLocalUnixAddress(grpc_endpoint* /*ep*/) { return "unix:"; }
+
+const grpc_endpoint_vtable kUnixEndpointVtable = {
+    nullptr, nullptr, nullptr, nullptr,
+    nullptr, nullptr, nullptr, GetLocalUnixAddress,
+    nullptr, nullptr};
+
+absl::string_view GetLocalTcpAddress(grpc_endpoint* /*ep*/) {
+  return "ipv4:127.0.0.1:12667";
+}
+
+const grpc_endpoint_vtable kTcpEndpointVtable = {
+    nullptr, nullptr, nullptr, nullptr,
+    nullptr, nullptr, nullptr, GetLocalTcpAddress,
+    nullptr, nullptr};
+
+void CheckSecurityLevelForServer(grpc_local_connect_type connect_type,
```

Comment:
The pattern of putting expectations inside helper functions makes tests much harder to debug, because when the expectation fails, the line number of the failure makes it hard to see exactly which test is failing.  Instead, I suggest changing these functions to return the actual security level, and then having the test case set the expectation on the returned value.